### PR TITLE
Write snapshot when SNAPSHOT_DIR is set

### PR DESCRIPTION
**Pull request by madninja**

### CODE REVIEW

File: src/be_db_block.erl [hunk 1]
```diff
@@ -12,138 +12,200 @@
...
+            lager:warning("Failed to write snapshot ~p: ~p", [Height, {What, Why, Where}])
+    end.
+
+maybe_write_snapshot(_, _, false, _Chain) ->
+    ok;
+maybe_write_snapshot(_, <<>>, _, _Chain) ->
+    ok;
+maybe_write_snapshot(Height, SnapshotHash, SnapshotDir, Chain) ->
+    {ok, Snapshot} = blockchain:get_snapshot(SnapshotHash, Chain),
+    Filename = filename:join([SnapshotDir, io_lib:format("snap-~p", [Height])]),
```
Comment (abhay) [inline src/be_db_block.erl:141]: add network name here?

File: src/be_db_block.erl [hunk 2]
```diff
@@ -12,138 +12,200 @@
...
+    {ok, BinSnap} = blockchain_ledger_snapshot_v1:serialize(Snapshot),
+    ok = file:write_file(Filename, BinSnap).
```
Comment (abhay) [inline src/be_db_block.erl:143]: will this end up being a directory that fills up forever? note for our future selves that we should have disk monitoring.

File: src/be_db_block.erl (hunk 1 above)
Comment (abhay) [reply src/be_db_block.erl:141]: is there any diagnostic value in adding a b58 / b64url encoded file for people to verify before they download?

File: src/be_db_block.erl (hunk 2 above)
Comment (madninja) [reply src/be_db_block.erl:142]: heh... yes it will

File: src/be_db_block.erl (hunk 1 above)
Comment (madninja) [reply src/be_db_block.erl:141]: I put the name in the SNAPSHOT_DIR env variable... e.g. `/snaps/mainnet`

**Pull request merged**